src/oscar/apps/search/search_indexes.py

```diff
@@ -106,3 +106,6 @@ def get_updated_field(self):
         Can be used to filter the query set when updating the index
         """
         return 'date_updated'
+
+    def should_update(self, instance, **kwargs):
+        return instance.is_public and not instance.is_child
```

Comment:
What is the reason you need to exclude these products from the index? I know you don't want these on the frontend, but you can just filter them out? For backend search application you still need them to be indexed.